### PR TITLE
Fix/2349  activation of share feature per workspace

### PR DESCRIPTION
**Pull request by inkhey**
related to #2349

## Checkpoints

The goal of this section is to help code reviewer to do required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

- [x] Code is clear enough
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

### CODE REVIEW

**Comment by inkhey** [comment]
in draft because waiting for frontend dev.

File: backend/tracim_backend/models/context_models.py [hunk 1]
```diff
@@ -681,6 +694,20 @@ def agenda_enabled(self) -> bool:
...
+        """
+        return self.workspace.public_download_enabled
+
+    @property
+    def public_upload_enabled(self) -> bool:
+        """
+        returns True if public upload is enabled in this workspace
+        """
+        return self.workspace.public_upload_enabled
+
```
Comment (buxx) [inline backend/tracim_backend/models/context_models.py]: Where is used these two perperties ? Are they really required ?

Comment (buxx) [reply backend/tracim_backend/models/context_models.py]: (If not useful, remove it. After that this merge request is mergeable)

Comment (inkhey) [reply backend/tracim_backend/models/context_models.py]: These are workspace properties, this allow to allow/disallow public download (content share) or public upload (upload permission) on specific workspace.

Comment (buxx) [reply backend/tracim_backend/models/context_models.py]: Okay, i missready github diff render :p

File: frontend_app_file/src/container/File.jsx [hunk 2]
```diff
@@ -778,111 +778,119 @@ class File extends React.Component {
...
-            lastModification={displayDistanceDate(state.content.modified, state.loggedUser.lang)}
-            lastModificationFormatted={(new Date(state.content.modified)).toLocaleString(props.i18n.language)}
-            description={state.content.raw_content}
-            displayChangeDescriptionBtn={state.loggedUser.userRoleIdInWorkspace >= 2}
-            disableChangeDescription={!state.content.is_editable}
-            onClickValidateNewDescription={this.handleClickValidateNewDescription}
-          />
-        }
-      ]
+    if (state.config.workspace.downloadEnabled && state.loggedUser.userRoleIdInWorkspace > 1) {
```
Comment (Skylsmoi) [inline frontend_app_file/src/container/File.jsx]: fix indentation in this function

File: frontend_app_file/src/container/File.jsx [hunk 3]
```diff
@@ -778,111 +778,119 @@ class File extends React.Component {
...
-            lastModification={displayDistanceDate(state.content.modified, state.loggedUser.lang)}
-            lastModificationFormatted={(new Date(state.content.modified)).toLocaleString(props.i18n.language)}
-            description={state.content.raw_content}
-            displayChangeDescriptionBtn={state.loggedUser.userRoleIdInWorkspace >= 2}
-            disableChangeDescription={!state.content.is_editable}
-            onClickValidateNewDescription={this.handleClickValidateNewDescription}
-          />
-        }
-      ]
+      return [{
```
Comment (Skylsmoi) [inline frontend_app_file/src/container/File.jsx]: in your "if" and your "else" there are 2 components that are the same (Timeline and FileProperty).

I haven't check if they have exactly the same props but if they have, put each array elements in a variable and only return an array of these variables. It'll save a LOT of lines of code.

File: frontend_app_file/src/container/File.jsx [hunk 4]
```diff
@@ -778,119 +778,78 @@ class File extends React.Component {
...
-                  lastModificationFormatted={(new Date(state.content.modified)).toLocaleString(props.i18n.language)}
-                  description={state.content.raw_content}
-                  displayChangeDescriptionBtn={state.loggedUser.userRoleIdInWorkspace >= 2}
-                  disableChangeDescription={!state.content.is_editable}
-                  onClickValidateNewDescription={this.handleClickValidateNewDescription}
-                  key={'FileProperties'}
-                />
-              )
-            }]
+      return [ timelineObject, {
```
Comment (Skylsmoi) [inline frontend_app_file/src/container/File.jsx]: put one element per lines please, it's more readable

**Comment by Skylsmoi** [comment]
@gbonaspetti I have accepted the review, my mistake, I'm actually requesting a small change

**Pull request merged**